src/GraphQL/Types/GraphType.cs

```diff
@@ -69,20 +68,7 @@ public string Name
 
         public string DeprecationReason { get; set; }
 
-        public virtual string CollectTypes(TypeCollectionContext context)
-        {
-            if (string.IsNullOrWhiteSpace(Name))
-            {
-                Name = GetType().Name;
```

Comment:
I've always been confused by this line of code - side effect when calling `CollectTypes`, setting `Name` property. I guessed that this line is never executed after we merged PR where each type has been assigned a default name in its constructor. I ran tests to check. And so it happened. In any case, I am against the side effects. If for some reason the name is not specified, then let it return `null` and you can then deal with the error instead of hiding it.